src/GitVersion.Core/VersionCalculation/TrunkBased/NonTrunk/FirstCommitOnRelease.cs

```diff
@@ -0,0 +1,34 @@
+using GitVersion.Configuration;
+
+namespace GitVersion.VersionCalculation.TrunkBased.NonTrunk;
+
+internal sealed class FirstCommitOnRelease : ITrunkBasedIncrementer
+{
+    // B  57 minutes ago  (HEAD -> release/1.0.0)
+    // A  58 minutes ago <<--
+
+    // A  58 minutes ago  (HEAD -> release/1.0.0) <<--
+
+    // C  56 minutes ago  (HEAD -> release/1.0.0)
+    // B  57 minutes ago <<--
+    // A  58 minutes ago (main)
```

Comment:
This was a bit too terse for me, can you please elaborate this? Perhaps by explaining the conditions inside the `MatchPrecondition` method?

Reply:
With this peace of code we identify the first commit on a branch marked with IsReleaseBranch true and append the version number e.g. 1.0.0 (extracted from the branch name) as an alternative semantic version. This will be used later to bump the version number to a higher value if necessary.

Reply:
Right. Could you explain this inside the code somehow? As I mentioned, I don't think the provided comments give enough information.